x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/IndexRequestCreator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.inference.persistence;
+
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+import org.elasticsearch.xpack.core.ml.utils.ToXContentParams;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Set;
+
+public class IndexRequestCreator {
+    private static final ToXContent.Params FOR_INTERNAL_STORAGE_PARAMS = new ToXContent.MapParams(
+        Collections.singletonMap(ToXContentParams.FOR_INTERNAL_STORAGE, "true")
+    );
+
+    private static final Set<String> PRE_PACKAGED_MODELS = Set.of(".elser_model_1");
+
+    public static IndexRequest create(String modelId, String docId, String index, ToXContentObject body) {
```

Comment:
Another idea I had here was this method to accept an interface that looks something like:

```
interface Identifiable {
  String getModelId();
  String getDocId();
}
```

I wasn't sure if that was overkill or didn't make sense to have `getDocId()` always defined for classes like `TrainedModelConfig`. I suppose I could extend the class to include that functionality.